### PR TITLE
CI/PR build improvements

### PR DESCRIPTION
**Pull request by rainersigwald**
Various improvements:

* Actually run tests
* Do debug & release
* Some general cleanup

### CODE REVIEW

File: .vsts-dotnet-ci.yml [hunk 1]
```diff
@@ -1,29 +1,36 @@
...
   "Parameters.solution": MSBuildLocator.sln
-  BuildConfiguration: Release
   SignType: Test
-  BuildPlatform: ''
 
 steps:
-- task: MSBuild@1
-  displayName: Set version number
-  condition: eq(variables['system.pullrequest.isfork'], false)
+- task: DotNetCoreInstaller@0
```
Comment (BenVillalobos) [inline .vsts-dotnet-ci.yml]: I can't seem to find notes on `DotNetCoreInstaller` in the [tasks documentation](https://docs.microsoft.com/en-us/azure/devops/pipelines/tasks/index?view=azure-devops)

Comment (BenVillalobos) [reply .vsts-dotnet-ci.yml]: Should this possibly be UseDotNet?

https://docs.microsoft.com/en-us/azure/devops/pipelines/tasks/tool/dotnet-core-tool-installer?view=azure-devops

Comment (rainersigwald) [reply .vsts-dotnet-ci.yml]: Yup, looks like a recent rename according to https://github.com/microsoft/azure-pipelines-tasks/pull/10044#issuecomment-481055325.

This does work: https://github.com/microsoft/azure-pipelines-tasks/tree/0a63e0e6c725bc7aeb6609e07123843267ef3a7e/Tasks/DotNetCoreInstallerV0

but yeah, I'll move over to the new one.

**Pull request merged**